plugins/scaffolder-react/api-report.md

```diff
@@ -459,18 +459,28 @@ export type ScaffolderTaskStatus =
 // @public
 export interface ScaffolderUseTemplateSecrets {
   // (undocumented)
-  secrets: Record<string, string>;
+  secrets: Secrets;
   // (undocumented)
-  setSecrets: (input: Record<string, string>) => void;
+  setSecrets: (input: SetSecretsInput) => void;
 }
 
+// @public
+export type Secrets = {
+  [key: string]: string | Secrets | (string | Secrets)[];
+};
+
 // @public
 export const SecretsContextProvider: (
   props: PropsWithChildren<{
     initialSecrets?: Record<string, string>;
   }>,
 ) => React_2.JSX.Element;
 
+// @public
+export type SetSecretsInput = {
```

Comment:
Not sure that it's worth us exporting these types to be honest, it's more API surface, and  I think that we should just change the type from `Record<string, string>` to `JsonObject` instead.

Those types are compatible too, so it's not breaking either.

Same for `Secrets`too

Reply:
Hm, I seem to have a problem with JsonObject - JsonValue doesn't allow 'undefined' and I explicitly need that for merging array values with lodash

```
merge({ arr: ["foo"] }, { arr: [undefined, "bar"] }) // => ["foo", "bar"]
merge({ arr: ["foo"] }, { arr: [null, "bar"] }) // => ["foo", null, "bar"]
```

Reply:
That's why the SetSecretsInput type is different and includes undefined - maybe you have a different idea?

Reply:
Actually - nevermind. I was able to fix it.